### PR TITLE
Feature/frontend/profile page

### PR DESCRIPTION
**Pull request by serkan-ozel-2015400123**
Initial profile, settings and timeline pages
Loading controller dismisses after response or after error, 10 s waiting bug resolve.
Adjusting routes accordingly
Event module is moved inside profile module to load them altogether as they would be always used interchangebly

### CODE REVIEW

**Comment by nazmicancalik** [comment]
Good job

**Comment by nazmicancalik** [comment]
We should merge this before 20.11.2018 so that we can test before lecture. @okyksl @kemaltulum @dkilyas @didemongu

File: frontend/src/app/profile/pages/settings/settings.page.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  }
+  changeCity(newCity : string){
+
+    let newUser : User = {
+      name: this.user.name,
+      email: this.user.email,
+      profileImage: this.user.profileImage,
+      _id: this.user._id,
+      birth: this.user.birth,
```
Comment (nazmicancalik) [inline frontend/src/app/profile/pages/settings/settings.page.ts]: birth is added along with some other fields to a new encapsulating field in the user data model. Please update the fields accordingly. Good job. 👍

**Comment by serkan-ozel-2015400123** [comment]
@kemaltulum
1-  I get first name and last name from backend instead of just name.
2- I have just implemented that part now and you can review again now. Not tested though

**Comment by kemaltulum** [comment]
I debugged the code
In settings page, your changes do not go to backend. The reason is that:
 you get the user object from backend:

``` 
_v: 0
​
_id: "5bf538fb76054d002f3a0877"
​
email: "deneme@deneme.com"
​
followers: Array []
​
following: Array []
​
interests: Array []
​
password: "$2b$10$/.u9.R4H16Vz55i5KlOMfO1v1Jy7aMq59oV7Ue.6CiKqdhRZl6UFC"
```

this do not contain fields like name, birth etc. Therefore, two-way binding fails

**Comment by serkan-ozel-2015400123** [comment]
Currently I am waiting put user endpoint to be tested because it doesnt work in my tests, so dont merge now

File: frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
-];
+  { path: 'feed', loadChildren: './profile/events/events.module#EventsModule'},
+  { path: 'profile', loadChildren: './profile/profile.module#ProfileModule'},
+  { path: '', loadChildren: './auth/auth.module#AuthModule', pathMatch: 'full'},
+  { path: '**', component: NotFoundPage}];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forRoot(routes,
+    { enableTracing : false , preloadingStrategy: PreloadAllModules})],
```
Comment (okyksl) [inline frontend/src/app/app-routing.module.ts]: Why did we choose `PreloadAllModules`? What does it do?

**Comment by serkan-ozel-2015400123** [comment]
> Why do we have events module folder inside the profile module folder?

It's because you have said it might be good to have events module inside profile since they will be used interchangebly.

**Comment by okyksl** [comment]
@serkan-ozel-2015400123 I meant to merge the two modules into _one module_ since they seem to be closely related. We can have separate or merged modules, but _a module inside a module_ is a bit confusing. Actually, module inside a module can be utilized in cases where there are strict encapsulation.

**Comment by serkan-ozel-2015400123** [comment]
@okyksl Okay I will merge them ASAP

**Comment by okyksl** [comment]
@kemaltulum @serkan-ozel-2015400123 there was a bug concerning with `user.userDetails.name` and there have been some changes:

* `name` is accessible from the **root** object.
* `userDetails` is renamed to `details`.

**Comment by serkan-ozel-2015400123** [comment]
When I sign up backend saves my user as userDetails{ name: "..."} . Why this is not fixed?

**Comment by serkan-ozel-2015400123** [comment]
I have changed settings file accordingly, but my testing is not possible  due to that

**Comment by serkan-ozel-2015400123** [comment]
@bounswe/bounswe2018group9 Need review here. Because backend and deployment issues this pr took too long to merge. I have implemented it with dummy data right now and it is working. Connecting to backend is to be  done.

**Comment by serkan-ozel-2015400123** [comment]
[HPM] Error occurred while trying to proxy request /api/auth/signin from localhost:4200 to http://46.101.223.116:3000 (ECONNREFUSED) (https://nodejs.org/api/errors.html#errors_common_system_errors)

 This is the error I take regarding proxy

**Pull request merged**